core/templates/pages/exploration-editor-page/exploration-save-and-publish-buttons/exploration-save-and-publish-buttons.component.ts

```diff
@@ -129,13 +131,33 @@ angular.module('oppia').component('explorationSaveAndPublishButtons', {
         $scope.saveIsInProcess = false;
         $scope.publishIsInProcess = false;
         $scope.loadingDotsAreShown = false;
+        ctrl.saveBeforeUnload();
+      };
+
+      $scope.canPublish = false;
 
+      $scope.showPublishButton = function() {
         UserExplorationPermissionsService.getPermissionsAsync()
           .then(function(permissions) {
-            $scope.showPublishButton = function() {
-              return permissions.canPublish && $scope.isPrivate();
-            };
```

Comment:
Define this function outside and use it in ng-if*

Reply:
isn't `showPublishButton` already outside ` UserExplorationPermissionsService.getPermissionsAsync`
rather UserExplorationPermissionsService.getPermissionsAsync` is inside `showPublishButton`

Reply:
(This comment looks old and related to some old commit and I'm not sure of the context here!) Earlier the `showPublishButton` was not an async function but it's async now. 

1. Try to avoid attaching such an async function to the scope.
2. The name of this function doesn't seems relevant anymore, right?

Reply:
[Line 146](https://github.com/oppia/oppia/pull/11690/files#diff-506d577c073a1d17c807d93d1ffa773e3c49589a4d8fc3844f14862120cd70baR146) must be a scope since the user permissions retrieved directly effect the frontend